src/Open3D/Visualization/Rendering/Filament/LineSetBuffers.cpp

```diff
@@ -0,0 +1,226 @@
+                    return values[i] < other.values[i];
+                }
+            }
+
+            return false;
+        }
+
+        const double kEpsilon = 0.00001;
+        double values[6] = {0};
+    };
+
+    // <source, real>
+    std::map<LookupKey, std::pair<GeometryBuffersBuilder::IndexType,
+                                  GeometryBuffersBuilder::IndexType>>
+            indexLookup;
+
+    const size_t linesCount = geometry_.lines_.size();
+    const size_t verticesBytesCount = linesCount * 2 * sizeof(ColoredVertex);
+    auto* vertices = static_cast<ColoredVertex*>(malloc(verticesBytesCount));
+
+    const size_t indicesBytesCount = linesCount * 2 * sizeof(IndexType);
+    auto* indices = static_cast<IndexType*>(malloc(indicesBytesCount));
+
+    const bool hasColors = geometry_.HasColors();
+    Eigen::Vector3d kWhite(1.0, 1.0, 1.0);
+    size_t vertexIndex = 0;
+    for (size_t i = 0; i < linesCount; ++i) {
+        const auto& line = geometry_.lines_[i];
+
+        for (size_t j = 0; j < 2; ++j) {
+            size_t index = line(j);
+
+            auto& color = kWhite;
+            if (hasColors) {
+                color = geometry_.colors_[i];
+            }
+            const auto& pos = geometry_.points_[index];
+
+            LookupKey lookupKey(pos, color);
+            auto found = indexLookup.find(lookupKey);
+            if (found != indexLookup.end()) {
+                index = found->second.second;
+            } else {
+                auto& element = vertices[vertexIndex];
+
+                element.SetVertexPosition(pos);
+                element.SetVertexColor(color);
+
+                indexLookup[lookupKey] = {index, vertexIndex};
+                index = vertexIndex;
+
+                ++vertexIndex;
+            }
+
+            indices[2 * i + j] = index;
+        }
+    }
+
+    const size_t verticesCount = vertexIndex;
+
+    VertexBuffer* vbuf = VertexBuffer::Builder()
+                                 .bufferCount(1)
+                                 .vertexCount(verticesCount)
+                                 .attribute(VertexAttribute::POSITION, 0,
+                                            VertexBuffer::AttributeType::FLOAT3,
+                                            ColoredVertex::GetPositionOffset(),
+                                            sizeof(ColoredVertex))
+                                 .normalized(VertexAttribute::COLOR)
+                                 .attribute(VertexAttribute::COLOR, 0,
+                                            VertexBuffer::AttributeType::FLOAT4,
+                                            ColoredVertex::GetColorOffset(),
+                                            sizeof(ColoredVertex))
+                                 .build(engine);
+
+    VertexBufferHandle vbHandle;
+    if (vbuf) {
+        vbHandle = resourceManager.AddVertexBuffer(vbuf);
+    } else {
+        free(vertices);
+        free(indices);
```

Comment:
I'm not sure that unique_ptr simplifies the logic, any, though. And "release" is a pretty crappy name for "transfer". I tend to defer to the original author here, and since I'm not confident on the proper way to test changes, I think we should defer any changes until after the release.